compiler/rustc_borrowck/src/type_check/mod.rs

```diff
@@ -2319,7 +2319,62 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
                         let cast_ty_from = CastTy::from_ty(ty_from);
                         let cast_ty_to = CastTy::from_ty(*ty);
                         match (cast_ty_from, cast_ty_to) {
-                            (Some(CastTy::Ptr(_)), Some(CastTy::Ptr(_))) => (),
+                            (Some(CastTy::Ptr(src)), Some(CastTy::Ptr(dst))) => {
+                                let src_tail = tcx.struct_tail_without_normalization(src.ty);
```

Comment:
Why no normalization?

Reply:
I think this is a mistake, I'll add normalization here

Reply:
OK -- it may be a bit tricky. If you need help LMK.

I think https://doc.rust-lang.org/nightly/nightly-rustc/rustc_borrowck/type_check/struct.TypeChecker.html#method.normalize should be sufficient (in the old trait solver).